torch/utils/data/datapipes/map/combinatorics.py

```diff
@@ -1,17 +1,16 @@
 import random
 
-from torch.utils.data.datapipes._decorator import functional_datapipe
-from torch.utils.data.datapipes.datapipe import MapDataPipe
+import torch
+from torch.utils.data.datapipes.datapipe import IterDataPipe, MapDataPipe
 from typing import Iterator, List, Optional, TypeVar
 
-__all__ = ["ShufflerMapDataPipe", ]
+__all__ = ["ShufflerIterDataPipe", ]
 
 
 T_co = TypeVar('T_co', covariant=True)
 
 
-@functional_datapipe('shuffle')
-class ShufflerMapDataPipe(MapDataPipe[T_co]):
+class ShufflerIterDataPipe(IterDataPipe[T_co]):
```

Comment:
```suggestion
# @functional_datapipe('shuffle')
class ShufflerIterDataPipe(IterDataPipe[T_co]):
```
Sorry for the annoying addition.

We need that extra commented out line in order to automatically generate the method interface for `MapDataPipe`.

Then we need to replace line 205 of `torch/utils/data/datapipes/gen_pyi.py` from:
`mapDP_method_to_special_output_type: Dict[str, str] = {}`
To:
`mapDP_method_to_special_output_type: Dict[str, str] = {"shuffle": "IterDataPipe"}`

Reply:
Thanks for mentioning it! Will add it.

This might be another BE task for PYI gen. lol

Reply:
Yea, the most ideal case is for `register_datapipe_as_function` to handle it.

At the same time, we generally don't plan to add more DataPipes (especially ones that require special treatments), so the cost/benefit tradeoff makes it very low priority.